### PR TITLE
Removing old InventoryCollection definitions

### PR DESCRIPTION
**Pull request by slemrmartin**
Also unifying spec IC model_class testing for EmbeddedAnsible

### CODE REVIEW

**Comment by slemrmartin** [comment]
Cc @jameswnl @agrare 
*There is cyclic dependency with core*

**Comment by Ladas** [comment]
@slemrmartin hm the dependency is not merged but the CI is passing?

File: spec/support/ansible_shared/automation_manager/refresher.rb [hunk 1]
```diff
@@ -237,7 +237,7 @@ def assert_inventory_root_group
     expect(expected_inventory_root_group).to have_attributes(
       :name    => "hello_inventory",
       :ems_ref => "99",
-      :type    => "ManageIQ::Providers::AutomationManager::InventoryRootGroup",
+      :type    => manager_class::InventoryRootGroup.name,
```
Comment (agrare) [inline spec/support/ansible_shared/automation_manager/refresher.rb]: See https://github.com/ManageIQ/manageiq/pull/17574#issuecomment-403505086 for why I don't like just changing the spec test

**Comment by slemrmartin** [comment]
@agrare now this PR doesn't depend on core

**Comment by slemrmartin** [comment]
no, it has to be merged together, because deleted files here will be missing in core before core's PR will be merged too. 
https://github.com/ManageIQ/manageiq/pull/17574

**Pull request merged**